astro/src/components/GoogleMap.vue

```diff
@@ -0,0 +1,197 @@
+<script setup lang="ts">
+import { onMounted, ref } from "vue"
+// @ts-ignore
+import { Loader } from "@googlemaps/js-api-loader"
+import './gmap.d.ts'
+
+import { t } from "i18next"
+
+interface Props {
+  coords: {
+    lat: number,
+    lng: number
+  },
+  details: {
+    city: string,
+    zip: string
+  },
+  address: string
+}
+
+const props = defineProps<Props>()
+const emit = defineEmits(["update:coords", "update:details", "update:address"])
+
+let GMAP_API_KEY = "AIzaSyAsg25KoeTUKIX_cuHAx4GWoLgESDRGgnU"
```

Comment:
remove this. you need to set it via env variable, never push keys to repositories